ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp

```diff
@@ -26,19 +27,18 @@ namespace operations::conv {
 namespace conv2d {
 
 struct Conv2dConfig {
-    MathFidelity math_fidelity = MathFidelity::HiFi4;
     DataType dtype = DataType::BFLOAT16;
     DataType weights_dtype = DataType::BFLOAT16;
-    bool math_approx_mode_enabled = true;
-    bool fp32_dest_acc_enabled = false;
-    bool packer_l1_accum_enabled = false;
     string activation = "";
     uint32_t input_channels_alignment = 32;
     bool deallocate_activation = false;
     bool reallocate_halo_output = false;
-    uint32_t act_block_h_override = 0; // This argument is ignored when shard_layout == WIDTH_SHARDED.
-    uint32_t act_block_w_div = 1; //Amount by which the maximum possible act_block_width is divided. Max act_block_w = (in_channels * window_w * window_h)/total_num_cores;
-                                  //Ignored when shard_layout == HEIGHT_SHARDED or BLOCK_SHARDED
+
+    //Amount by which the maximum possible act_block is divided along a certain dimension.
+    //Divided along height for Height & Block Sharding. Divided along width for Width Sharding.
+    //When not set, chooses the smallest possible act block size(lowest perf). Setting to 1 chooses largest possible (highest perf)
+    std::optional<uint32_t> act_block_div = std::nullopt;
```

Comment:
it's odd that having 
- nullopt means optimize memory
- 1 means optimize perf
- 2 means actually use 2
Seem like we want another filed here which is true/false (optimize mem vs perf), and other optional filed which like 
```suggestion
    std::optional<uint32_t> act_block_div = std::nullopt;
```
ignore the first option I want exact value.